netbox/netbox/api/authentication.py

```diff
@@ -123,6 +134,33 @@ def has_object_permission(self, request, view, obj):
 
         return super().has_object_permission(request, view, obj)
 
+    # Helper function to obtain the desired action from the first permission in the permission map
+    def get_action(self, method):
+        # Get the first permission
+        permission = self.perms_map.get(method)[0] if len(self.perms_map.get(method)) > 0 else None
+        if permission:
+            # Remove app and model label
+            action = permission.replace('%(app_label)s.', '').replace('_%(model_name)s', '')
```

Comment:
We can use `utilities.permissions.resolve_permission()` to get the action from the permission string.

Reply:
Thanks, seems like you always have a helper for something.  😄